src/mds/StrayManager.cc

```diff
@@ -300,6 +301,11 @@ void StrayManager::enqueue(CDentry *dn, bool trunc)
   CInode *in = dnl->get_inode();
   ceph_assert(in);
 
+  //remove inode from scrub stack if it is being purged
+  if(mds->scrubstack->remove_inode_if_stacked(in)) {
```

Comment:
How about

```
if (obj->item_scrub.is_on_list()) {
    remove from scrub stack
}
```

Reply:
so this would require the function `remove_inode_if_stacked` moved here, im fine with it. the reason i kept it in ScrubStack is because the function shares its relevance with ScrubStack